x-pack/plugins/enterprise_search/KEA.md

```diff
@@ -0,0 +1,178 @@
+}
+
+interface CustomSource {
+  name: string;
+}
+
+interface AddCustomSourceValues {
+  sourceApiStatus: ApiStatus<CustomSource>;
+}
+
+export const AddCustomSourceLogic = kea<
+  MakeLogicType<AddCustomSourceValues, AddCustomSourceActions>
+>({
+  path: ['enterprise_search', 'workplace_search', 'add_custom_source_logic'],
+  actions: {
+    fetchSource: true,
+    fetchSourceError: (code, error) => ({ code, error }),
+    fetchSourceSuccess: (customSource) => customSource,
+  },
+  reducers: () => ({
+    sourceApiStatus: [
+      {
+        status: 'IDLE',
+      },
+      {
+        fetchSource: () => ({
+          status: 'PENDING',
+        }),
+        fetchSourceError: (_, error) => ({
+          status: 'ERROR',
+          error,
+        }),
+        fetchSourceSuccess: (_, data) => ({
+          status: 'SUCCESS',
+          data,
+        }),
+      },
+    ],
+  }),
+  listeners: ({ actions }) => ({
+    fetchSource: async () => {
+      clearFlashMessages();
+
+      try {
+        const response = await HttpLogic.values.http.post<CustomSource>('/api/source');
+        actions.fetchSourceSuccess(response);
+      } catch (e) {
+        flashAPIErrors(e);
+        actions.fetchSourceError(e.code, e.message);
+      }
+    },
+  }),
+});
+```
+
+The types used above can be found in our [common Enterprise Search types file](common/types/api.ts). While the above assumes a single, idempotent API, this approach can be easily extended to use a dictionary approach:
+```
+reducers: () => ({
+    sourceApiStatus: [
+      {
+      },
+      {
+        fetchSource: (state, id) => ({...state,
+        id: {
+          status: 'PENDING',
+          data: state[id]?.data,
+        }}),
+        fetchSourceError: (_, ({id, error})) => ({...state,
+        id: {
+          status: 'ERROR',
+          error,
+        }}),
+        fetchSourceSuccess: (_, ({id, data})) => ({...state, id: {
+          status: 'SUCCESS',
+          data,
+        }}),
+      },
+    ],
+  }),
+```
+## Import actions and values from API logic files into component and view logic.
+
+Once you have an API interactions file set up, components and other Kea logic files can import the values from those files to build their own logic. Use the Kea 'connect' functionality to do this, as the auto-connect functionality has a few bugs and was removed in Kea 3.0. This allows you to read the status and value of an API, react to any API events, and abstract those APIs away from the components. Those components can now become more functional and reactive to the current state of the application, rather than to directly responding to API events.
+
+```
+export const AddCustomSourceLogic = kea<
+  MakeLogicType<AddCustomSourceValues, AddCustomSourceActions, AddCustomSourceProps>
+>({
+  connect: {
+    actions: [AddCustomSourceApiLogic, ['addSource', 'addSourceSuccess', 'addSourceError']],
+    values: [AddCustomSourceApiLogic, ['sourceApi']],
```

Comment:
I think we can add an example of how to add 'selective' connects from multiple Logic files. It is not immediately clear how to do it right on kea docs, I had to check source to figure. 

```javascript
actions: [
+      OtherLogic1,
+      ['action1', 'action2'],
+      OtherLogic2,
+      ['action3', 'action4'],
+    ],
```

Reply:
Sure!